src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -725,6 +725,10 @@ public function getSession()
             $this->setSession($session = $session());
         }
 
+        if (null === $session) {
+            @trigger_error(sprintf('Calling "%s()" when "%s::hasSession()" returns false is deprecated since Symfony 4.1 and will throw an exception in 5.0.', __METHOD__, __CLASS__), E_USER_DEPRECATED);
```

Comment:
Calling "%s()" when non session has been set is deprecated since Symfony 4.1 and will throw an exception in 5.0. Use "hasSession()" instead.

also, please add the exception tight now, but as a commented line, so that upgrading for 5.0 is made trivial:
`// throw new ...`

Reply:
Message updated.

About the `// throw new ...` I thought a type hinting on the return type with `SessionInterface` was enough, PHP will throw : 

> TypeError : Return value of Request::getSession() must implement interface SessionInterface.

WDYT ?

Reply:
I think we can provide a friendlier message, don't you?

Reply:
Updated